srcpkgs/tetgen/template

```diff
@@ -0,0 +1,25 @@
+# Template file for 'tetgen'
+pkgname=tetgen
+version=1.5.1
+revision=1
+build_style="gnu-makefile"
+make_cmd="make -C ${pkgname}${version} -f makefile"
+hostmakedepends="tar"
+short_desc="Quality Tetrahedral Mesh Generator and 3D Delaunay Triangulator"
+maintainer="Phicem <phicem@gmx.com>"
+license="AGPL-3.0-or-later"
+homepage="https://www.wias-berlin.de/software/index.jsp?id=TetGen"
+distfiles="http://www.dhondt.de/cgx_2.20.all.tar.bz2"
```

Comment:
this should probably be a subpackage of cgx

Reply:
Tetgen is actually a separate program, with different authors. It is distributed by tetgen website, but it is necessary to provide an email address before getting the download link. Tetgen is also distributed by CalculiX authors, this time with no email constraint. This is why I picked CalculiX website as the tetgen source.
What do you mean by "subpackage"? Should I call the tetgen package 'CalculiX-tetgen' ?

Reply:
Yeah, we don't play this game with Void packages.

You get the tarball from the proper upstream source, unless it is explicitly vendored into the rest of CalculiX, in which case it wouldn't be a separate package at all.

If this is difficult or impossible, it militates strongly against packaging, not grabbing upstream distfiles from some other distributor.

Reply:
Tetgen is indeed officially vendored along with CalculiX (more precisely with the cgx part). It is explicitely written in the documentation.

Initially I wanted to put tetgen a separate package in case someone wants to use tetgen without installing CalculiX-cgx. But it is probably a rare use-case, and Calculix-cgx package is not that huge (15 MB).

If that sounds good to you, I will include tetgen binary directly into CalculiX-cgx package.

Reply:
However if I do that (including tetgen binary directly into CalculiX-cgx package), I'm afraid I will have to define a custom do_build() function, since there are two independant makefile/cmake files. I could patch cgx makefile so that it calls tetgen makefile, but I don't find this very elegant.

An alternative is to keep tetgen in a separate sub-package called CalculiX-tetgen, using the tetgen version vendored by CalculiX.

In short, the solutions I see are:
1. single package with a custom do_build() function
2. single package with a patch on the makefile
3. keeping a separate subpackage called Calculix-tetgen

I don't know which one is the cleanest option. What do you think?

Reply:
If you need to give your email to access the download, it sounds likely to me that the licence might also be a problem in itself

Reply:
The authors say that tetgen is distributed "as Free Software under the [GNU Affero Public License v. 3.0](http://www.gnu.org/licenses/agpl-3.0.html)". But they also sell a commercial license for those who need it.

The GNU Affero Public License v. 3.0 says that "You are not required to accept this License in order to receive or run a copy of the Program", which is contradictory with the fact that they ask the user to give an email address and accept the license before getting the download link. So my understanding is that they don't really respect the license themselves (but I guess this is not a problem since they are the authors).

Reply:
Yeah it's not a problem for them if they are the copyright holders of 100% of the code (and I assume they are, in order to dual-licence).

Anyway it's some form of GPL so my potential concerns about the licence are addressed

Reply:
"Vendored" in this context means that the second project is integrated into the source tree and built alongside it. It seems what CalculiX is doing is just co-distributing tetgen, not vendoring it---the tarball is included but the CalculiX build process has no knowledge of this.

If you can resolve the licensing issue and get the original tarball from the original tetgen source, a separate package is OK. Otherwise, build it into the CalculiX package. You can override `do_build` or define `pre_build` or `post_build` as appropriate.

I don't see much point in making tetgen a subpackage, and that doesn't resolve the build override problem anyway---subpackage installation steps are not meant to add additional build steps, just to split installed contents.

Reply:
Okay, thank you for the detail. Tetgen is now built directly in CalculiX-cgx package.